python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -48,6 +49,78 @@ def destroy_engine(self):
         if self.msg_server is not None:
             self.msg_server.close()
 
+
+    ###########################################################################
+    # panel support (displayed as tabs)
+    def _init_tabs(self):
+        """
+        Create desktop tabs from the registered application panels, following
+        their specified relative positioning.
+        """
+        # make the "Apps" tab appear as a fake panel to fit in the sorting
+        # logic
+        apps_tab = {"callback":   self.desktop_window._register_apps_tab,
+                    "properties": {"position": 0}}
+
+        panels = [apps_tab] + list(self._engine.panels.itervalues())
+
+        def panel_position(panel):
+            """ Returns the relative position of a panel """
+            # sane checking on the expected structure
+            assert("properties" in panel, "panel should have properties set")
+            assert("position" in panel["properties"],
+                   "panel should have a position set")
+            return panel["properties"]["position"]
+
+        ordered_panels = sorted(panels, key=panel_position)
```

Comment:
@robblau Oh, I misunderstood the intended behaviour. Will update.
As for the "automatic" ordering of unspecified positions, how about an `auto_position` property with the automatic numbering? Otherwise we'll get surprising default ordering since the panels are stored in a dictionary with their names as keys (i.e. they're not in the same order as the registers happened).
(We could also store panels in an `OrderedDict` instead, but that would be in `tk-core`)